solr/solrj/src/test/org/apache/solr/client/solrj/impl/CloudHttp2SolrClientTest.java

```diff
@@ -262,12 +263,18 @@ public void testHttpCSPPerf() throws Exception {
     httpBasedCloudSolrClient.add(COLLECTION, doc);
     httpBasedCloudSolrClient.commit(COLLECTION);
 
+    Pattern patternWithCollection =
+        Pattern.compile(
+            "path=/admin/collections.*params=\\{[^}]*action=CLUSTERSTATUS[^}]*collection=[^&}]+[^}]*\\}");
+
+    Pattern patternWithoutCollection =
+        Pattern.compile(
+            "path=/admin/collections.*params=\\{[^}]*action=CLUSTERSTATUS(?![^}]*collection=)[^}]*\\}");
```

Comment:
Okay.   Albeit, these regex's are obtuse.  It'd be nice if LogListener had a `predicate` method taking `Predicate<String>` then you'd have something more readable like contains this and doesn't contain that.  Ah well.  Feel free to ignore this comment.